src/applications/disability-benefits/526EZ/helpers.jsx

```diff
@@ -93,6 +93,47 @@ export const getReservesGuardData = formData => {
   };
 };
 
+/**
+ * Converts the treatment date range into an array of objects from just an object
+ * @param {object} treatmentDateRange object containing from/to date range
+ * @returns {array} of treatmentDateRange's
+ */
+
+const transformDateRange = treatmentDateRange => [treatmentDateRange];
+
+/**
+ * Cycles through the list of provider facilities and performs transformations on each property as needed
+ * @param {array} providerFacility array of objects being transformed
+ * @returns {object} containing the new Provider Facility structure
+ */
+const transformProviderFacility = providerFacility => {
+  const newProviderFacility = [];
+  providerFacility.forEach(facility => {
+    newProviderFacility.push({
+      providerFacilityName: facility.providerFacilityName,
+      treatmentDateRange: transformDateRange(facility.treatmentDateRange),
+      providerFacilityAddress: facility.providerFacilityAddress,
+    });
+  });
+
+  return newProviderFacility;
+};
+
+/**
+ * If any limited consent text is populated, collect it.
+ */
+export function gatherLimitedConsentText(disabilities) {
+  const fullLimitedConsent = disabilities
+    .filter(disability => disability['view:limitedConsent'])
+    .reduce((accumulator, disability) => {
+      let string = accumulator;
+      string = `${string} ${disability.limitedConsent}`;
```

Comment:
@Jennywalter Correct me if I am wrong here, but I believe that we do not want the limited consent to be correlated to a specific condition.

Reply:
Correct.  If you look at the actual 4142 form, Limited Consent is captured only once and it is the way a Veteran can indicate he/she would like to limit the VA from retrieving certain medical details, not necessarily for a specific contention.  We spoke to the business and it was decided to capture Limited Consent at an aggregate level in the UI and not at a specific contention or provider level.

Reply:
@igreulich there is no real limit to the number of characters we can take on the pdf.  anything over 83 characters will go to the overflow pages in full.

Reply:
Cool, that makes sense. I think it might be clearer to separate each limited consent entry by a new line that way the single entry that gets put in the PDF isn't a potentially confusing mash of entries.